Library/ViewModels/PledgeViewModel.swift

```diff
@@ -74,15 +74,20 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
           .materialize()
       }
 
-    self.presentShippingRules = shippingRulesEvent.values()
-
-    let selectedShippingRule = Signal.merge(
+    let defaultShippingRule = Signal.merge(
       self.viewDidLoadProperty.signal.mapConst(nil),
-      self.presentShippingRules.map(defaultShippingRule(fromShippingRules:))
+      shippingRulesEvent.values().map(defaultShippingRule(fromShippingRules:))
+    )
+
+    self.presentShippingRules = Signal.combineLatest(
+      project,
+      shippingRulesEvent.values(),
+      defaultShippingRule.skipNil()
     )
+    .takeWhen(self.shippingRuleSignal.signal)
```

Comment:
`shippingRuleSignal` doesn't seem like the right name here, this probably just points to our delegate naming perhaps not being explanatory enough. I think we'd want this to indicate that the `selectShippingRuleButton` was tapped and that the currently selected `ShippingRule` was passed in.

So perhaps the delegate should be:

```swift
func pledgeShippingCellWillPresentShippingRules(_: PledgeShippingLocationCell, selectedShippingRule rule: ShippingRule)
// or
func pledgeShippingCellPresentedShippingRules(_: PledgeShippingLocationCell, selectedShippingRule rule: ShippingRule)
```
I think we can then base the view model input on that and it should improve the signal naming.